### PR TITLE
[AND-147] - Implement auto-update worker

### PR DESCRIPTION
**Pull request by GuiFPires**
**What does this PR do?**

It makes AG autoupdate all the apps that are installed by it every 24 hours (automatically), if the user has the auto-update setting on.
It also makes the getMeta calls on InstallPackageInfoMapper optional so we don't make getMeta requests when we auto-update, as we already have the needed information.

**Database changed?**

No

**Where should the reviewer start?**

- [ ] See by commit

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [AND-147](https://aptoide.atlassian.net/browse/AND-147)

**What are the relevant tickets?**

  Tickets related to this pull-request: [AND-147](https://aptoide.atlassian.net/browse/AND-147)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[AND-147]: https://aptoide.atlassian.net/browse/AND-147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AND-147]: https://aptoide.atlassian.net/browse/AND-147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by phillipsgoncalves — APPROVED** [review]
approved with the app in background. I will test in the weekend with the normal case of having it dead.

File: feature_updates/src/main/java/cm/aptoide/pt/feature_updates/domain/Updates.kt [hunk 1]
```diff
@@ -109,4 +117,23 @@ class Updates @Inject constructor(
...
+        if (VERSION.SDK_INT >= VERSION_CODES.UPSIDE_DOWN_CAKE) {
+          packageManager.getInstallSourceInfo(it.packageName).updateOwnerPackageName == applicationContext.packageName
+        } else if (VERSION.SDK_INT >= VERSION_CODES.R) {
+          packageManager.getInstallSourceInfo(it.packageName).installingPackageName == applicationContext.packageName
+        } else {
+          false
+        } || it.packageName == applicationContext.packageName
+      }
+      .forEach {
+        installManager.getApp(it.packageName).install(
```
Comment (pedroagostinhoaptoide) [inline feature_updates/src/main/java/cm/aptoide/pt/feature_updates/domain/Updates.kt:134]: This should use constraints for Wifi only downloads

**Review by phillipsgoncalves — APPROVED** [review]

File: feature_updates/src/main/java/cm/aptoide/pt/feature_updates/domain/Updates.kt (hunk 1 above)
Comment (GuiFPires) [reply feature_updates/src/main/java/cm/aptoide/pt/feature_updates/domain/Updates.kt:136]: Fixed

File: app-games/src/main/java/com/aptoide/android/aptoidegames/updates/repository/UpdatesPreferencesRepository.kt
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/updates/repository/UpdatesPreferencesRepository.kt:1]: I believe this logic should be in the `feature_updates` module

File: feature_updates/src/main/java/cm/aptoide/pt/feature_updates/domain/Updates.kt [hunk 2]
```diff
@@ -109,4 +119,27 @@ class Updates @Inject constructor(
...
```
Comment (G0retZ) [inline feature_updates/src/main/java/cm/aptoide/pt/feature_updates/domain/Updates.kt:136]: What if user already have tasks in a queue that will run before auto-update tasks? 🤔

File: feature_updates/src/main/java/cm/aptoide/pt/feature_updates/data/AutoUpdateWorker.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    fun enqueue(context: Context) = WorkManager.getInstance(context)
+      .enqueueUniquePeriodicWork(
+        WORK_NAME,
+        KEEP,
+        PeriodicWorkRequestBuilder<AutoUpdateWorker>(Duration.ofHours(24))
+          .setInitialDelay(Duration.ofHours(24))
+          .setConstraints(
+            Constraints.Builder()
+              .setRequiredNetworkType(UNMETERED)
```
Comment (G0retZ) [inline feature_updates/src/main/java/cm/aptoide/pt/feature_updates/data/AutoUpdateWorker.kt:40]: We already have a worker that runs on network changes. I wonder how they will play together? 🤔

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/updates/repository/UpdatesPreferencesRepository.kt
Comment (phillipsgoncalves) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/updates/repository/UpdatesPreferencesRepository.kt:1]: I agree, even though this way we will be forcing everyone that uses the feature_updates module to have the worker starting in the beginning (which might not be the behavior they want) and have the same worker durations/schedules.
Even though, if this is a problem in the future, then we can make it configurable, for example.

**Pull request merged**